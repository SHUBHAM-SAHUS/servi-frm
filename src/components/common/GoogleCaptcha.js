import
React,
{
    Component
} from 'react';
import {
    loadReCaptcha,
    ReCaptcha
} from 'react-recaptcha-google';

import { SITEKEY } from '../../config.js';

class GoogleCaptcha extends Component {
    constructor(props, context) {
        super(props, context);
        this.onLoadRecaptcha = this.onLoadRecaptcha.bind(this);
        this.verifyCallback = this.verifyCallback.bind(this);
    }
  
    componentDidMount() {
        loadReCaptcha();
        if (this.captchaDemo) {
            this.captchaDemo.reset();
        }
    }
   
    onLoadRecaptcha() {
        if (this.captchaDemo) {
            this.captchaDemo.reset();
        }
    }
   
    verifyCallback(recaptchaToken) {
        if (recaptchaToken) {
            this.props.handlerFromParant("Verified");
        }
    }

    expiredCallback() {
        this.captchaDemo.reset();
    }

    render() {
        return (
            <div>
                <ReCaptcha
                    ref={(el) => { this.captchaDemo = el; }}
                    size="normal"
                    data-theme="dark"
                    render="explicit"
                    data-badge="null"
                    sitekey= {SITEKEY}
                    onloadCallback={this.onLoadRecaptcha}
                    verifyCallback={this.verifyCallback}
                    expiredCallback={this.expiredCallback}
                />
            </div>
        );
    };
};

export default GoogleCaptcha;