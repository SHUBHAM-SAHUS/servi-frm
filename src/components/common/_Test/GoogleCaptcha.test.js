import React from 'react';
import { shallow } from 'enzyme';

import GoogleCaptcha from '../GoogleCaptcha';
describe('GoogleCaptcha', () => {
  it('should render correctly GoogleCaptcha', () => {
    const component = shallow(<GoogleCaptcha />);
  
    expect(component).toMatchSnapshot();
  });
});