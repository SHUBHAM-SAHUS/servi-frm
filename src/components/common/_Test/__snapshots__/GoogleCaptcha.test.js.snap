// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoogleCaptcha should render correctly GoogleCaptcha 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GoogleCaptcha />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ReCaptcha
        badge="bottomright"
        data-badge="null"
        data-theme="dark"
        elementID="g-recaptcha"
        expiredCallback={[Function]}
        expiredCallbackName="expiredCallback"
        hl="en"
        onloadCallback={[Function]}
        onloadCallbackName="onloadCallback"
        render="explicit"
        sitekey="6Le1aqkUAAAAAAPsov0VYB2-9L5xrsfyJJ9FnoYA"
        size="normal"
        tabindex="0"
        theme="light"
        type="image"
        verifyCallback={[Function]}
        verifyCallbackName="verifyCallback"
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "badge": "bottomright",
        "data-badge": "null",
        "data-theme": "dark",
        "elementID": "g-recaptcha",
        "expiredCallback": [Function],
        "expiredCallbackName": "expiredCallback",
        "hl": "en",
        "onloadCallback": [Function],
        "onloadCallbackName": "onloadCallback",
        "render": "explicit",
        "sitekey": "6Le1aqkUAAAAAAPsov0VYB2-9L5xrsfyJJ9FnoYA",
        "size": "normal",
        "tabindex": "0",
        "theme": "light",
        "type": "image",
        "verifyCallback": [Function],
        "verifyCallbackName": "verifyCallback",
      },
      "ref": [Function],
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ReCaptcha
          badge="bottomright"
          data-badge="null"
          data-theme="dark"
          elementID="g-recaptcha"
          expiredCallback={[Function]}
          expiredCallbackName="expiredCallback"
          hl="en"
          onloadCallback={[Function]}
          onloadCallbackName="onloadCallback"
          render="explicit"
          sitekey="6Le1aqkUAAAAAAPsov0VYB2-9L5xrsfyJJ9FnoYA"
          size="normal"
          tabindex="0"
          theme="light"
          type="image"
          verifyCallback={[Function]}
          verifyCallbackName="verifyCallback"
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "badge": "bottomright",
          "data-badge": "null",
          "data-theme": "dark",
          "elementID": "g-recaptcha",
          "expiredCallback": [Function],
          "expiredCallbackName": "expiredCallback",
          "hl": "en",
          "onloadCallback": [Function],
          "onloadCallbackName": "onloadCallback",
          "render": "explicit",
          "sitekey": "6Le1aqkUAAAAAAPsov0VYB2-9L5xrsfyJJ9FnoYA",
          "size": "normal",
          "tabindex": "0",
          "theme": "light",
          "type": "image",
          "verifyCallback": [Function],
          "verifyCallbackName": "verifyCallback",
        },
        "ref": [Function],
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
